src/platform/telink/wifi/3.7.99.0/WiFiManager.cpp

```diff
@@ -193,8 +193,18 @@ CHIP_ERROR WiFiManager::Scan(const ByteSpan & ssid, ScanResultCallback resultCal
     wifi_scan_params * scanParams{ nullptr };
     size_t scanParamsSize{ 0 };
 
-    if (!ssid.empty())
+    if (!ssid.empty()) // Directed Scanning, Main Spec, 11.9.7.1. "ScanNetworks Command"
     {
+        if (!mInternalScan) // don't erase in the middle of Connect procedure
+        {
+            Instance().mDirectedScanning = true;
+            mWantedNetwork.Erase();
+            memcpy(mWantedNetwork.ssid, ssid.data(), ssid.size());
+            mWantedNetwork.ssidLen = ssid.size();
+            ChipLogProgress(DeviceLayer, "Directed Scanning, looking for: %.*s",
```

Comment:
same comments as above.
Are these files a copy & paste? Could they share code?

Reply:
For now we have to mantain both Zephyr version so we keep them in spare files. May be refactored in future